pkg/querier/remote_read_test.go

```diff
@@ -20,30 +20,70 @@ import (
 	"github.com/prometheus/common/model"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/storage"
+	prom_remote "github.com/prometheus/prometheus/storage/remote"
 	"github.com/stretchr/testify/require"
 
 	"github.com/grafana/mimir/pkg/ingester/client"
 	"github.com/grafana/mimir/pkg/mimirpb"
 	"github.com/grafana/mimir/pkg/storage/series"
 )
 
-func TestRemoteReadHandler(t *testing.T) {
-	q := storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
-		return mockQuerier{
-			matrix: model.Matrix{
-				{
-					Metric: model.Metric{"foo": "bar"},
-					Values: []model.SamplePair{
-						{Timestamp: 0, Value: 0},
-						{Timestamp: 1, Value: 1},
-						{Timestamp: 2, Value: 2},
-						{Timestamp: 3, Value: 3},
+type mockSampleAndChunkQueryable struct {
+	queryableFn      func(ctx context.Context, mint, maxt int64) (storage.Querier, error)
+	chunkQueryableFn func(ctx context.Context, mint, maxt int64) (storage.ChunkQuerier, error)
+}
+
+func (m mockSampleAndChunkQueryable) Querier(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
+	return m.queryableFn(ctx, mint, maxt)
+}
+
+func (m mockSampleAndChunkQueryable) ChunkQuerier(ctx context.Context, mint, maxt int64) (storage.ChunkQuerier, error) {
+	return m.chunkQueryableFn(ctx, mint, maxt)
+}
+
+type mockQuerier struct {
+	storage.Querier
+	matrix model.Matrix
+}
+
+func (m mockQuerier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
+	if sp == nil {
+		panic("mockQuerier: select params must be set")
+	}
+	return series.MatrixToSeriesSet(m.matrix)
+}
+
+type mockChunkQuerier struct {
+	storage.ChunkQuerier
+	matrix model.Matrix
+}
+
+func (m mockChunkQuerier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Matcher) storage.ChunkSeriesSet {
+	if sp == nil {
+		panic("mockChunkQuerier: select params must be set")
+	}
+	return storage.NewSeriesSetToChunkSet(series.MatrixToSeriesSet(m.matrix))
+}
+
+func TestSampledRemoteRead(t *testing.T) {
+	q := &mockSampleAndChunkQueryable{
+		queryableFn: func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
+			return mockQuerier{
+				matrix: model.Matrix{
+					{
+						Metric: model.Metric{"foo": "bar"},
+						Values: []model.SamplePair{
+							{Timestamp: 0, Value: 0},
+							{Timestamp: 1, Value: 1},
+							{Timestamp: 2, Value: 2},
+							{Timestamp: 3, Value: 3},
+						},
 					},
 				},
-			},
-		}, nil
-	})
-	handler := RemoteReadHandler(q, log.NewNopLogger())
+			}, nil
+		},
+	}
+	handler := RemoteReadHandler(q, 1024*1024, log.NewNopLogger())
```

Comment:
This hard codes `1024*1024` which is duplicate with it being hard-coded here: https://github.com/grafana/mimir/pull/1735/files#diff-9b906049355d90fe28012174be8ec0587a836c5bd0cd09dcbccc968bc481939aR162
```suggestion
	handler := RemoteReadHandler(q, 1024*1024, log.NewNopLogger())
```

I think this introduces a risk that if in the future the definition of this constant in the `api` package gets changed, that this unit test gets forgotten about and then we have a unit test which tests with a different setting.

Reply:
We got rid of the max frame size parameter at 6019c81913ae717ec6f6e660fbf0713d3a7914ca. So this no longer applies.